pkg/resource/deploy/step_generator.go

```diff
@@ -17,6 +17,7 @@ package deploy
 import (
 	cryptorand "crypto/rand"
 	"fmt"
+	"slices"
```

Comment:
Do we also need to be able to build `pkg` with Go 1.20? Looks like this package was added in Go 1.21: https://go.dev/doc/go1.21#slices

Reply:
Ah we don't run them on PR, but we run the integration tests that depend on pkg on 1.20 for merge. So I think this would have failed to merge if we tried now.